app/src/main/java/com/example/bottomnav/Parse.java

```diff
@@ -61,6 +103,11 @@ private void storeConst(CPPASTSimpleDeclaration declaration) throws Exception {
                     (CPPASTLiteralExpression) comp.init.get().getInitializerClause();
             ConstContents contents = new ConstContents(comp.name, value, comp.typeQualifier,
                     comp.type);
+            Pattern pattern = Pattern.compile("\\dx(\\d+)");
```

Comment:
Is there any way to get the value as an int out of the AST? Or some way for the AST to evaluate the literal value? The more the AST / C++ parsing libraries can do for us, the better - for example, if the value is `42` or `0x2a` - or maybe even `20+22`.

But if you do want to match, it should be as strict as possible. Hex numbers in C start with `0x`, you shouldn't match on any-digit-then-x.

Also, consider moving this into the ConstContents, perhaps having a ConstContents.decode static method that can reject bad values.